### ISSUE TITLE
Files uploaded to /public should not need a signed url to access

### ISSUE THREAD

**Issue by AlexThomas90210** [aws-amplify/amplify-js#659]
Do you want to request a feature or report a bug?
feature 

What is the current behaviour?
Currently, when using Storage.put it uploads files to the public/ directory of the user-files s3 bucket. These files require a signed URL to access.

What is the expected behaviour?
Currently, there are 3 directories in the user-files by default, they are
**/public**          -> files here should be accessible to everybody without a signed url 
**/protected**    -> these files should have the current functionality of what /public is nowadays, in that these files are accessible by any user from the Cognito user pool given they have signed the request
**/private**         -> these files should be only accessible via the person who actually uploaded the files

Making the /public folder accessible to everybody enables many common use cases which are currently hindered by needed to wait for the signed URL from storage.get. The main ones are:

Uploading profile photos of users.
Uploading banner images for user-generated content (eg the displayed image on youtube videos)
Basically, any website which makes high use of user-generated photos.

Then if the /public directory did not need a signed URL you could then easily configure it with your CloudFront distribution by adding a new origin. In fact, AWS Amplify & AWS Mobile-CLI could see if I have a CloudFront distribution for my project and automatically sync my user-generated /public origin to my CloudFront distribution

Additionally, it makes the code for displaying user-generated content much simpler as I don't need to wait for the Promise to finish from Storage.get for every single photo I am displaying before displaying it, which sometimes could be 20-40 images in one page

**Comment by mwarger** [comment]
Related to https://github.com/aws/aws-amplify/issues/518?

**Comment by thejuan** [comment]
The permissions when putting these files need to allow read by everyone

**Comment by sammartinez** [comment]
Resolving this Feature request in favor of #518
